youtube_dl/extractor/beampro.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+    ExtractorError,
+    compat_str,
+    clean_html,
+    parse_iso8601,
+)
+
+
+class BeamProLiveIE(InfoExtractor):
+    IE_NAME = 'Beam:live'
+    _VALID_URL = r'https?://(?:\w+.)?beam.pro/(?P<id>[^?]+)$'
+    _API_CHANNEL = 'https://beam.pro/api/v1/channels/{0}'
+    _API_MANIFEST = 'https://beam.pro/api/v1/channels/{0}/manifest.{1}'
+    _VALID_MANIFESTS = ('smil', 'm3u8', 'light', 'light2', 'ftl', 'ftlOld')
+    _RATINGS = {'family': 0, 'teen': 13, '18+': 18}
+
+    _TEST = {
+        'url': 'http://www.beam.pro/niterhayven',
+        'info_dict': {
+            'id': '261562',
+            'ext': 'mp4',
+            'uploader': 'niterhayven',
+            'timestamp': 1483477281,
+            'age_limit': 18,
+            'title': 'Introducing The Witcher 3 //  The Grind Starts Now!',
+            'thumbnail': r're:https://.*\.jpg$',
+            'upload_date': '20170103',
+            'uploader_id': 373396,
+            'description': 'md5:0b161ac080f15fe05d18a07adb44a74d',
+            'is_live': True,
+        },
+        'skip': 'niterhayven is offline',
+        'params': {
+            'skip_download': True,
+        },
+    }
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+        chan_data = self._download_json(self._API_CHANNEL.format(channel_id), channel_id)
+
+        if not chan_data.get('online'):
```

Comment:
but it is:
 `info.get('online')` returns `None if 'online' doesn't exist and `if not None` is True in that case.